plugins/chat/assets/javascripts/discourse/components/chat-channel.js

```diff
@@ -107,7 +111,12 @@ export default class ChatLivePane extends Component {
     if (this._loadedChannelId !== this.args.channel?.id) {
       this._unsubscribeToUpdates(this._loadedChannelId);
       this.chatChannelPane.selectingMessages = false;
-      this.chatChannelComposer.cancelEditing();
+      this.chatChannelComposer.message =
+        this.args.channel.draft ||
+        ChatMessage.createDraftMessage(this.args.channel, {
+          user: this.currentUser,
+        });
+      this.chatChannelComposer.onCancelEditing();
```

Comment:
I think this is why the draft is not loading, since we set the draft above and then do onCancelEditing which goes to reset which does this:

```js
@action
reset() {
  this.message = ChatMessage.createDraftMessage(this.model, {
    user: this.currentUser,
  });
}
```